include/xen/generic.h

```diff
@@ -11,4 +11,27 @@
 #define XEN_PAGE_SIZE		4096
 #define XEN_PAGE_SHIFT		12
 
+#define PFN_UP(x)		(unsigned long)(((x) + XEN_PAGE_SIZE-1) >> XEN_PAGE_SHIFT)
+#define PFN_DOWN(x)		(unsigned long)((x) >> XEN_PAGE_SHIFT)
+#define PFN_PHYS(x)		((unsigned long)(x) << XEN_PAGE_SHIFT)
+#define PHYS_PFN(x)		(unsigned long)((x) >> XEN_PAGE_SHIFT)
+
+#define to_phys(x)		((unsigned long)(x))
+#define to_virt(x)		((void *)(x))
```

Comment:
Shouldn't those two helpers be in a non-Xen headers?

Reply:
Probably yes, but I did not find such in already implemented helpers in Zephyr headers. I left them in Xen header but also changed them to be less generic names.